substrate/frame/delegated-staking/src/impls.rs

```diff
@@ -0,0 +1,318 @@
+		T::CoreStaking::is_unbonding(who)
+	}
+
+	fn fully_unbond(who: &Self::AccountId) -> DispatchResult {
+		ensure!(Self::is_delegatee(who), Error::<T>::NotSupported);
+		return T::CoreStaking::fully_unbond(who);
+	}
+
+	fn bond(
+		who: &Self::AccountId,
+		value: Self::Balance,
+		payee: &Self::AccountId,
+	) -> DispatchResult {
+		// ensure who is not already staked
+		ensure!(T::CoreStaking::status(who).is_err(), Error::<T>::NotDelegatee);
+		let delegatee = Delegatee::<T>::from(who)?;
+
+		ensure!(delegatee.available_to_bond() >= value, Error::<T>::NotEnoughFunds);
+		ensure!(delegatee.ledger.payee == *payee, Error::<T>::InvalidRewardDestination);
+
+		T::CoreStaking::bond(who, value, payee)
+	}
+
+	fn nominate(who: &Self::AccountId, validators: Vec<Self::AccountId>) -> DispatchResult {
+		ensure!(Self::is_delegatee(who), Error::<T>::NotSupported);
+		return T::CoreStaking::nominate(who, validators);
+	}
+
+	fn chill(who: &Self::AccountId) -> DispatchResult {
+		ensure!(Self::is_delegatee(who), Error::<T>::NotSupported);
+		return T::CoreStaking::chill(who);
+	}
+
+	fn bond_extra(who: &Self::AccountId, extra: Self::Balance) -> DispatchResult {
+		let delegation_register = <Delegatees<T>>::get(who).ok_or(Error::<T>::NotDelegatee)?;
+		ensure!(delegation_register.stakeable_balance() >= extra, Error::<T>::NotEnoughFunds);
+
+		T::CoreStaking::bond_extra(who, extra)
+	}
+
+	fn unbond(stash: &Self::AccountId, value: Self::Balance) -> DispatchResult {
+		let delegatee = Delegatee::<T>::from(stash)?;
+		ensure!(delegatee.bonded_stake() >= value, Error::<T>::NotEnoughFunds);
+
+		T::CoreStaking::unbond(stash, value)
+	}
+
+	/// Withdraw unbonding funds until current era.
+	///
+	/// Funds are moved to unclaimed_withdrawals register of the `DelegateeLedger`.
+	fn withdraw_unbonded(
+		pool_acc: Self::AccountId,
+		num_slashing_spans: u32,
+	) -> Result<bool, DispatchError> {
+		Pallet::<T>::withdraw_unbonded(&pool_acc, num_slashing_spans)
+			.map(|delegatee| delegatee.ledger.total_delegated.is_zero())
+	}
+
+	fn desired_validator_count() -> u32 {
+		defensive_assert!(false, "not supported for delegated impl of staking interface");
+		T::CoreStaking::desired_validator_count()
+	}
+
+	fn election_ongoing() -> bool {
+		defensive_assert!(false, "not supported for delegated impl of staking interface");
+		T::CoreStaking::election_ongoing()
+	}
+
+	fn force_unstake(_who: Self::AccountId) -> DispatchResult {
+		defensive_assert!(false, "not supported for delegated impl of staking interface");
+		Err(Error::<T>::NotSupported.into())
+	}
+
+	fn is_exposed_in_era(who: &Self::AccountId, era: &EraIndex) -> bool {
+		defensive_assert!(false, "not supported for delegated impl of staking interface");
+		T::CoreStaking::is_exposed_in_era(who, era)
+	}
+
+	fn status(who: &Self::AccountId) -> Result<StakerStatus<Self::AccountId>, DispatchError> {
+		ensure!(Self::is_delegatee(who), Error::<T>::NotSupported);
```

Comment:
I put defensive asserts when those functions should not really be called since pool does not need it while e2e testing. But I think I can remove most of them now.